usaspending_api/references/v2/views/agency.py

```diff
@@ -46,11 +47,21 @@ def get(self, request, pk, format=None):
         # need to filter on
         # (used filter() instead of get() b/c we likely don't want to raise an
         # error on a bad agency id)
-        queryset = queryset.filter(
-            submission__reporting_fiscal_year=active_fiscal_year,
-            submission__reporting_fiscal_quarter=active_fiscal_quarter,
-            treasury_account_identifier__funding_toptier_agency=toptier_agency
-        )
+        # DS-1655: if the AID is "097" (DOD), Include the branches of the military in the queryset
+        if toptier_agency and toptier_agency.cgac_code and toptier_agency.cgac_code == "097":
```

Comment:
Why are these conditional statements necessary? `if toptier_agency and toptier_agency.cgac_code`

above that we have the following:
```
         agency = Agency.objects.filter(id=agency_id).first()
 
         if agency is None:
             return Response(response)
 
          toptier_agency = agency.toptier_agency
```
So if we get to the later lines of code, we assume we have a valid agency. With that, are those checks still necessary?

Reply:
Their are empty agencies in the db (from when we didnt run the cleanup script), which can cause these ifs to fail

Reply:
So it's possible for `toptier_agency = agency.toptier_agency` to be `None`? If that's the case, this is a larger issue since it _shouldn't_ be the case if it's being inserted into `Agency` and `ToptierAgency`

Reply:
woops thats an award bug not an agency bug...